### ISSUE TITLE
[Bug]: Accessibility issue: maximum-scale=1 added to viewport meta tag in the `<head>` prevents zooming

### ISSUE THREAD

**Issue by alionazherdetska** [storybookjs/storybook#31272]
### Describe the bug

Storybook adds `maximum-scale=1` to the viewport meta tag in the HTML head of the documentation pages, which disables zooming functionality. This creates a critical accessibility issue as users with visual impairments who need to zoom cannot do so.
This prevents compliance with **WCAG 2.1 Success Criterion 1.4.4 (Resize text)**, which requires that text can be resized without assistive technology up to 200 percent without loss of content or functionality.

![Image](https://github.com/user-attachments/assets/dc021074-6896-442b-84ed-d58de8da3d29)

### Reproduction link

https://temporary-design-system-post.netlify.app/?path=/docs/introduction--docs

### Reproduction steps

1. Go to https://temporary-design-system-post.netlify.app/?path=/docs/introduction--docs
2. Right-click and inspect the page, then look at the <head> section
3. Check the meta tag: `<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">` inside `<head>`

### System

```bash
Storybook version 8.2.7
```

**Comment by Sidnioulz** [comment]
Just commenting to confirm this is a serious accessibility issue. The `max-zoom` should not be used. Be mindful that it was added to circumvent a UI bug, which will also need proper fixing.

The code was initially added by @cdedreuille to "fix an issue when clicking on search" as part of https://github.com/storybookjs/storybook/pull/24228. Charles, if you happen to remember the exact issue you had at the time, feel free to comment!

**Comment by Sidnioulz** [comment]
@ghengeveld assigning you on the issue too so it shows up on the associated GH project and so I know someone owns it 🙏
